include/lsst/afw/table/misc.h

```diff
@@ -20,6 +20,11 @@ namespace lsst { namespace afw { namespace table {
  */
 typedef boost::int64_t RecordId;
 
+enum CoordinateType {
+    PIXEL,
+    CELESTIAL
+};
+
```

Comment:
Could this be an `enum class` instead?  It looks like gcc 4.4 supports those, though I'm not sure if Swig does.  Looks like it's worth just trying it; your unit tests should be sufficient to see if it works.